internal/clusterfeature/features/monitoring/operator.go

```diff
@@ -0,0 +1,638 @@
+		if existingSecretID != "" {
+			logger.Debug("Prometheus secret already exists")
+			return prometheusSecretName, nil
+		} else if isSecretNotFoundError(err) {
+			// generate and store Prometheus secret
+			secretName, err = op.generatePrometheusSecret(ctx, cluster)
+			if err != nil {
+				return "", errors.WrapIf(err, "failed to generate Prometheus secret")
+			}
+		} else {
+			return "", errors.WrapIf(err, "error during getting Prometheus secret")
+		}
+	} else {
+		var err error
+		secretName, err = op.secretStore.GetNameByID(ctx, spec.Prometheus.SecretId)
+		if err != nil {
+			return "", errors.WrapIfWithDetails(err, "failed to get Prometheus secret", "secretID", spec.Prometheus.SecretId)
+		}
+	}
+	return secretName, nil
+}
+
+func (op FeatureOperator) ensureOrgIDInContext(ctx context.Context, clusterID uint) (context.Context, error) {
+	if _, ok := auth.GetCurrentOrganizationID(ctx); !ok {
+		cluster, err := op.clusterGetter.GetClusterByIDOnly(ctx, clusterID)
+		if err != nil {
+			return ctx, errors.WrapIf(err, "failed to get cluster by ID")
+		}
+		ctx = auth.SetCurrentOrganizationID(ctx, cluster.GetOrganizationId())
+	}
+	return ctx, nil
+}
+
+func (op FeatureOperator) generateAlertManagerProvidersConfig(spec providerSpec) configValues {
+	return configValues{
+		Global: configGlobalValues{
+			Receivers: []receiverItemValues{
+				{
+					Name:             alertManagerProviderConfigName,
+					SlackConfigs:     op.generateSlackConfig(spec.Slack),
+					EmailConfigs:     op.generateEmailConfig(spec.Email),
+					PagerdutyConfigs: op.generatePagerdutyConfig(spec.Pagerduty),
+				},
+			},
+		},
+	}
+}
+
+func (op FeatureOperator) generateSlackConfig(config slackPropertiesSpec) []slackConfigValues {
+	if config.Enabled {
+		return []slackConfigValues{
+			{
+				ApiUrl:       config.ApiUrl,
+				Channel:      config.Channel,
+				SendResolved: config.SendResolved,
+			},
+		}
+	}
+
+	return nil
+}
+
+func (op FeatureOperator) generateEmailConfig(config emailPropertiesSpec) []emailConfigValues {
+	if config.Enabled {
+		return []emailConfigValues{
+			{
+				To:           config.To,
+				From:         config.From,
+				SendResolved: config.SendResolved,
+			},
+		}
+	}
+
+	return nil
+}
+
+func (op FeatureOperator) generatePagerdutyConfig(config pagerdutyPropertiesSpec) []pagerdutyConfigValues {
+	if config.Enabled {
+		return []pagerdutyConfigValues{
+			{
+				RoutingKey:   config.RoutingKey,
+				ServiceKey:   config.ServiceKey,
+				Url:          config.Url,
+				SendResolved: config.SendResolved,
+			},
+		}
+	}
+
+	return nil
+}
+
+func getPrometheusOperatorChartName() string {
+	return viper.GetString(config.PrometheusOperatorChartKey)
+}
+
+func getPrometheusOperatorVersion() string {
+	return viper.GetString(config.PrometheusOperatorVersionKey)
+}
+
+func getPrometheusPushgatewayChartParams() (name, version string) {
+	name = viper.GetString(config.PrometheusPushgatewayChartKey)
```

Comment:
I'd pass config values (or config value provider(s)) to the operator instead of depending on `viper` here. I'd also consider making separate methods/functions for getting the two params.